spec/controllers/api/v1/organizations_controller_spec.rb

```diff
@@ -0,0 +1,133 @@
+require 'spec_helper'
+
+describe Api::V1::OrganizationsController, type: :controller do
+  let(:authorized_user) { create(:user) }
+  let(:organization) { build(:organization, listed_at: Time.now, owner: authorized_user) }
+  let(:unlisted_organization) { build(:organization, listed_at: nil) }
+
+  let(:scopes) { %w(public organization) }
+
+  describe "when not logged in" do
+    describe "#index" do
+
+      it 'returns only the listed organization' do
+        unlisted_organization.save
+        organization.save
+        get :index
+        expect(response.status).to eq(200)
+        expect(json_response["organizations"].length).to eq(1)
+      end
+    end
+  end
+
+  describe "when a logged in user" do
+    describe '#index' do
+      it_behaves_like "is indexable" do
+        let(:private_resource) { unlisted_organization }
+        let(:api_resource_name) { 'organizations' }
+        let(:api_resource_attributes) { %w(id name display_name) }
+        let(:api_resource_links) { %w() }
+
+        let(:n_visible) { 1 }
+
+        before do
+          organization.save
+          private_resource.save
+        end
+      end
+
+      it 'returns listed organizations' do
+        organization.save
+        get :index
+        expect(response.status).to eq(200)
+        expect(json_response['organizations'].map { |o| o['id'] }).to include(organization.id.to_s)
+      end
+
+      describe "with unlisted organizations" do
+        let(:owned_unlisted_organization) { build(:organization, listed_at: nil, owner: authorized_user) }
+
+        before do
+          unlisted_organization.save
+          owned_unlisted_organization.save
+        end
+
+        it 'returns unlisted organizations that I own' do
+          default_request scopes: scopes, user_id: authorized_user.id
+          get :index
+          expect(json_response['organizations'].map { |o| o['id'] }).to include(owned_unlisted_organization.id.to_s)
+        end
+
+        it "doesn't return unlisted organizations for unauthorized users" do
```

Comment:
Goal was to ensure that logged in users didn't see other users' unlisted orgs. Fixed.